common/src/test/scala/uk/ac/wellcome/platform/archive/common/generators/IngestGenerators.scala

```diff
@@ -45,7 +45,17 @@ trait IngestGenerators extends BagIdGenerators {
     )
 
   def createIngestEvent: IngestEvent =
-    IngestEvent(randomAlphanumeric)
+    createIngestEventWith()
+
+  def createIngestEventWith(
+    description: String = randomAlphanumeric,
+    createdDate: Instant =
+      Instant.now().plusSeconds(randomInt(from = 0, to = 30))
+  ): IngestEvent =
+    IngestEvent(
+      description = description,
+      createdDate = createdDate
+    )
```

Comment:
Unless I'm mistaken, won't these default parameters just get initialised once, rather than randomly at each call?

Assuming that is not what you want, maybe would make sense for  the function to take `Option[_]` instead, and default to `None`, e.g.

```scala
def createIngestEventWith(
  description: Option[String] = None,
  createdDate: Option[Instant] = None
    
): IngestEvent =
  IngestEvent(
    description = description.getOrElse(randomAlphanumeric),
    createdDate = createdDate.getOrElse(
      Instant.now.plusSeconds(randomInt(from = 0, to = 30))
    )
  )

```

Reply:
Perhaps surprisingly, no! (I had to double-check this – and thanks for making me check. I’ve seen what looks like the correct behaviour in the past, but never checked explicitly.)

Here's a little repro (from master, so it's just got the `randomAlphanumeric` bit):

```scala
import uk.ac.wellcome.platform.archive.common.generators.IngestGenerators
import uk.ac.wellcome.platform.archive.common.ingests.models.IngestEvent

object Experiments extends IngestGenerators {
  def x(): Seq[IngestEvent] = {
    Seq(createIngestEvent, createIngestEvent, createIngestEvent)
  }
}

println(Experiments.x())

// List(
//   IngestEvent(ImogKzTd,2019-07-29T11:09:22.099Z),
//   IngestEvent(NK1WKolY,2019-07-29T11:09:22.099Z),
//   IngestEvent(TvfYL6CJ,2019-07-29T11:09:22.099Z)
// )
```

According to the [Scala docs](https://docs.scala-lang.org/sips/completed/named-and-default-arguments.html) about default arguments:

> A method parameter with a default argument has the form `"p: T = expr"`, where `expr` is evaluated every time a method application uses the default argument

Reply:
:+1: ah that is good to know that it evaluates the expression each time for default args, no what I was expecting